libs/event-manager/web-client/data-access/src/lib/event-api.service.ts

```diff
@@ -173,18 +191,46 @@ export class EventApiService {
     });
   }
 
-  findByTicketOwner(owner: PublicKey) {
+  // find published event by firebase id
+  findByFId(eventFId: string): Observable<EventAccount | undefined> {
+    return defer(() => {
+      const reader = this.reader;
+
+      if (reader === null) {
+        return throwError(() => new Error('ProgramReaderMissing'));
+      }
+
+      return from(
+        reader.account.event
+          .all()
+          .then((events) => events as EventAccount[])
+          .then((events: EventAccount[]) =>
+            events.find((event) => event.account.fId === eventFId)
```

Comment:
This query is very expensive. We'd have to get all events from the blockchain, then filter the whole response. If we used the firebase id to generate the PDA of an event, we'd turn this query into O(1).